playbooks/roles/start/tasks/wait_for_zowe.yml

```diff
@@ -99,6 +109,33 @@
   until: _result.status == 204
   retries: "{{ wait_for_zowe_service_retries | int }}"
   delay: "{{ wait_for_zowe_service_delay | int }}"
+  # if zowe_zos_host and zowe_external_domain_name are not same, zlux could be
+  # started off z/OS and then we couldn't test right now
+  when: zowe_zos_host == zowe_external_domain_name and "api-mediation" in zowe_launch_components
+  delegate_to: localhost
+  # hide log to avoid exposing zowe_test_user and zowe_test_password
+  no_log: True
+
+- name: Wait for being able to login to zLux
+  uri:
+    url: "https://{{ zowe_zos_host }}:{{ zowe_zlux_port }}/auth"
+    follow_redirects: none
+    method: POST
+    body_format: json
+    body: 
+      username: "{{ zowe_test_user }}"
+      password: "{{ zowe_test_password }}"
+    validate_certs: false
+    status_code:
+    - 200
+    - 204
+  register: _result
+  until: _result.status == 200
+  retries: "{{ wait_for_zowe_service_retries | int }}"
+  delay: "{{ wait_for_zowe_service_delay | int }}"
+  # if zowe_zos_host and zowe_external_domain_name are not same, zlux could be
+  # started off z/OS and then we couldn't test right now
+  when: zowe_zos_host == zowe_external_domain_name and "app-server" in zowe_launch_components
```

Comment:
Please change to `when: zowe_zos_host == zowe_external_domain_name and ("app-server" in zowe_launch_components or zowe_launch_components == "")`